src/dynamic_policy_service/dynamic.policy.js

```diff
@@ -0,0 +1,165 @@
+    }
+    return null;
+}
+
+/**
+ * Verify the type of remember-me.
+ * @param {*} rememberMe 
+ */
+const validateRememberMe = (rememberMe) => {
+    if (typeof(rememberMe) !== "boolean") {
+        const errorMessage = `type of policy ${constants.WANTED_REMEMBER_ME} should be a boolean`;
+        throw new Error(errorMessage);
+    }
+}
+
+/**
+ * @desc validate authType is an array of boolean.
+ * @param {*} authTypes 
+ */
+const validateWantedAuthTypes = (authTypes) => {
+    if (!Array.isArray(authTypes)) {
+        const errorMessage = `type of ${constants.WANTED_AUTH_TYPES} should be an array of numbers`;
+        throw new Error(errorMessage);
+    }
+}
+
+/**
+ * 
+ * @param {*} wanted wanted is an array of attributes to be requested.
+ * @param {*} authTypes auth types represeents the authentication type to be used.
+ * @param {*} wantedRememberMe 
+ */
+const Policy = (
+    wanted,
+    authTypes,
+    wantedRememberMe = false,
+    extensions = [],
+) => {
+    // Perform some validation
+    const policy = {};
+    // Validate policy.
+    validateWanted(wanted);
+    policy[constants.WANTED] = wanted;
+    // Assert an array of numbers.
+    if (authTypes) {
+        validateWantedAuthTypes(authTypes);
+    }
+    policy[constants.WANTED_AUTH_TYPES] = (authTypes) ? authTypes : [];
+    // Assert boolean value
+    validateRememberMe(wantedRememberMe)
+    policy[constants.WANTED_REMEMBER_ME] = wantedRememberMe;
+
+    policy[constants.EXTENSIONS] = extensions;
+    return policy;
+}
+
+class DynamicPolicyRequest {
+    /**
+     * 
+     * @param {*} callbackEndpoint refers to the endpoint to callback to.
+     * @param {*} policy is the poilicy to be requested, this refers to thte attributes.
+     */
+    constructor(callbackEndpoint, policy) {
+        this.setCallBackEndpoint(callbackEndpoint);
+        this.setPolicy(policy);
+    }
+
+    /**
+     * @param {*} callbackEndpoint sets the callbackEndpoint
+     */
+    setCallBackEndpoint(callbackEndpoint) {
+        this.callbackEndpoint = callbackEndpoint;
+    }
+
+    /**
+     * 
+     * @param {*} policy sets the policy.
+     */
+    setPolicy(policy) {
+        const validatedPolicy = Policy(
+            policy[constants.WANTED],
+            policy[constants.WANTED_AUTH_TYPES],
+            policy[constants.WANTED_REMEMBER_ME],
+        );
+        this.policy = validatedPolicy;
+    }
+
+    /**
+     * @param {wantedPolicy} wantedPolicy is a policy to add.
+     */
+    addWantedPolicy(wantedPolicy) {
+        validateWanted(wantedPolicy);
+        this.policy.wanted.push(wantedPolicy);
+    }
+
+    /**
+     *  @returns a json serialised object request.
+     */
+    getData() {
+        const data = {};
+        data[constants.CALLBACK_ENDPOINT] = this.callbackEndpoint;
+        data[constants.POLICY] = this.policy;
+        data[constants.EXTENSIONS] = [];
+        return data;
+    }
+};
+
+
+/**
+ * @description is the response object from connect.
```

Comment:
Not sure if the same thing I mentioned above will apply here, as it's code and not documentation - referring to Connect